crates/typst/src/text/mod.rs

```diff
@@ -1300,3 +1304,24 @@ cast! {
         ret
     },
 }
+
+/// Function to parse the language argument.
+/// Provides a hint if a region is used in the language field.
+fn parse_lang(args: &mut Args) -> SourceResult<Option<Lang>> {
+    let Some(Spanned { v: iso, span }) = args.named::<Spanned<EcoString>>("lang")? else {
+        return Ok(None);
+    };
+
+    let result = Lang::from_str(&iso);
+    if result.is_err() {
+        if let Some((lang, region)) = iso.split_once('-') {
+            if Lang::from_str(lang).is_ok() && Region::from_str(region).is_ok() {
+                return result
+                    .hint(eco_format!("you should use the `region` field for '{region}'"))
```

Comment:
```suggestion
                    .hint(eco_format!("you should use the `region` parameter for '{region}'"))
```
The documentation typically refers to those as "parameters" I think.

Reply:
I think we should also use double quotes around `{region}` because otherwise it won't be valid Typst syntax.

Reply:
I have refined the hint. Pleace review it.